resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/FilterConfigSource.java

```diff
@@ -0,0 +1,62 @@
+package org.jboss.resteasy.microprofile.config;
+
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletContext;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.jboss.resteasy.core.ResteasyContext;
+
+public class FilterConfigSource implements ConfigSource {
+
+   private String name;
+
+   @Override
+   public Map<String, String> getProperties() {
+      FilterConfig config = ResteasyContext.getContextData(FilterConfig.class);
+      if (config == null) {
+         return Collections.<String, String>emptyMap();
+      }
+      Map<String, String> map = new HashMap<String, String>();
+      Enumeration<String> keys = config.getInitParameterNames();
+      for (String key = keys.nextElement(); keys.hasMoreElements(); key = keys.nextElement()) {
+         map.put(key, config.getInitParameter(key));
+      }
+      return map;
+   }
+
+   @Override
+   public String getValue(String propertyName) {
+      FilterConfig config = ResteasyContext.getContextData(FilterConfig.class);
+      if (config == null) {
+         return null;
+      }
+      return config.getInitParameter(propertyName);
+   }
+
+   @Override
+   public String getName() {
+      if (name == null) {
+         synchronized(this) {
+            if (name == null) {
```

Comment:
@ronsigal , shouldn't the "name" field be volatile for this to work properly?

Reply:
D'oh. I knew there was something bothering me about that. Fixed. Thanks.